journal/202308.md

```diff
@@ -0,0 +1,870 @@
+
+- Wait until all other startup tasks have been completed before listening to [notifications from dcrd](https://github.com/decred/vspd/pull/407). There was no problem with connecting to dcrd earlier but it was unnecessary and created messy logs.
+- Changed discovery of [voted and revoked tickets](https://github.com/decred/vspd/pull/413) from using dcrwallet's `TicketInfo` call to using [GCS filters](https://github.com/decred/dcrd/tree/f69247ff2478057ddbbc6692fc96ff6520b6a8c1/gcs) from dcrd. [Golomb-coded Sets](https://github.com/decred/dcps/blob/master/dcp-0005/dcp-0005.mediawiki#user-content-Golombcoded_Sets) (GCS) enable very efficient detection of transactions, this feature is a key element in Decred's privacy-preserving light wallets (SPV). The new method is [more robust](https://github.com/decred/vspd/pull/413#issuecomment-1694228745), and as a result VSP admins may notice that vspd detects some historic voted/revoked tickets which `TicketInfo` never detected.
+- Added a new integrity check that updates revoked tickets in the database to be either [expired or missed](https://github.com/decred/vspd/pull/415). Initial version of vspd was unable to distinguish between expired and missed tickets and reported them all as "revoked". Knowing how many tickets were missed by the VSP is useful for [evaluating its reliability](https://github.com/decred/vspd/issues/268). This update uses a clever heuristic to guess ticket outcome based on how long it took to revoke it. It is not 100% accurate but is good enough for the purpose of reporting VSP stats.
+
+Developer and internal changes:
+
+- Added test to ensure the hard-coded [error string](https://github.com/decred/vspd/pull/404) returned from [securecookie](https://github.com/gorilla/securecookie) will not change unexpectedly when vspd is upgraded to a newer version of the library.
+- [Replaced](https://github.com/decred/vspd/pull/401) uses of `interface{}` with `any`, similar to [dcrwallet](#dcrwallet).
+- Updated to [Go 1.21](https://github.com/decred/vspd/pull/402) and newer GitHub Actions. Testing with Go 1.19 has been removed.
+- Removed client-side [VSP fee payment logic](https://github.com/decred/vspd/pull/403) from vspd repository. The [idea](https://github.com/decred/vspd/issues/379) to migrate all VSP client code from dcrwallet to vspd repository was found to be problematic, it created a circular dependency between vspd and dcrwallet which is [cumbersome](https://matrix.to/#/!zefvTnlxYHPKvJMThI:decred.org/$_rzNyvnGedWeh_tnXWk5WDQKaj3TODeXbxRa9fg7CtY) and not worth the maintenance burden. Most VSP client code has been removed from vspd and will continue to live in dcrwallet repository. Eventually it should become an exported package which can be reused by software seeking to add VSP staking features.
+  - { _^ is it correct? What client VSP code remains in vspd then? And which module/package consumers like DEX or Cryptopower should use now to not copy-paste dcrwallet's VSP code?_ }
```

Comment:
This is totally correct. What remains in the vspd repo are the absolute basics which will be useful for any consumer of the vspd API:

- the `types` module -  the [data types](https://github.com/decred/vspd/blob/master/types/types.go) for communicating with vspd.
- the `client` module - a [basic HTTP client](https://github.com/decred/vspd/blob/master/client/client.go) which sends/receives those types to an instance of vspd, and includes signing requests/verifying signatures of responses.

To avoid the copy/pasting from dcrwallet, that code should be exported from the dcrwallet repo.

Reply:
Thanks for clearing that up. I got a bit confused across multiple PRs and issues about this in dcrwallet and vspd. Added a part of your comment, it's good dev content.